solr/core/src/java/org/apache/solr/core/SolrResourceLoader.java

```diff
@@ -201,24 +198,19 @@ void addToClassLoader(List<URL> urls) {
         .collect(Collectors.toList()));
   }
 
-  /**
-   * Adds URLs to the ResourceLoader's internal classloader.  This method <b>MUST</b>
-   * only be called prior to using this ResourceLoader to get any resources, otherwise
-   * its behavior will be non-deterministic. You also have to {link @reloadLuceneSPI}
-   * before using this ResourceLoader.
-   *
-   * @param urls    the URLs of files to add
-   */
-  void addToClassLoader(URL... urls) {
-    addToClassLoader(Arrays.asList(urls));
-  }
-  
   /**
    * Reloads all Lucene SPI implementations using the new classloader.
    * This method must be called after {@link #addToClassLoader(List)}
    * and before using this ResourceLoader.
    */
   void reloadLuceneSPI() {
+    // TODO improve to use a static Set<URL> to check when we need to
+    if (!needToReloadLuceneSPI) {
```

Comment:
is this thread safe? worst case we end up reloading more often than necessary, which isn't a big deal, right?

Reply:
Hey good point; needToReloadLuceneSPI should be volatile and reloadLuceneSPI() should be synchronized.  In practice it'd only happen if someone had some strange Solr hacks / plugins because the only callers of the couple methods that touch this boolean do so once and then never touch it again.  But still, lets be thorough.